sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -98,6 +130,19 @@ object ExtractIntervalPart {
     case "SECOND" | "S" | "SEC" | "SECONDS" | "SECS" => ExtractIntervalSeconds(source)
     case _ => errorHandleFunc
   }
+
+  def parseExtractFieldANSI(
```

Comment:
Can we merge this into `parseExtractField`?
```
case "YEAR" | "Y" | "YEARS" | "YR" | "YRS" =>
  if (source.dataType == YearMonthIntervalType) {
    ExtractANSIIntervalYears(source)
  } else {
    ExtractIntervalYears(source)
  }
```

Reply:
OK

Reply:
we need another branch for DayTimeIntervalType case